src/Nox.Solution/Models/Domain/Entities/Entity.cs

```diff
@@ -116,6 +122,33 @@ public virtual bool IsKey(string keyName)
         return _keysByName!.ContainsKey(keyName);
     }
 
+    public virtual bool TryGetRelationshipByName(NoxSolution solution, string relationshipName, out NoxSimpleTypeDefinition? result)
+    {
+        result = null;
+        if (Relationships == null)
+        {
+            return false;
+        }
+
+        var rel = Relationships!.First(x => x.Name.Equals(relationshipName));
+        // TODO: possibly extend for other types
+        if (!rel.ShouldGenerateForeignOnThisSide ||
```

Comment:
```suggestion
        if (!rel.ShouldGenerateForeignOnThisSide || rel.IsManyRelationshipOnOtherSide)
```

Reply:
Actually, this property is about a different thing, but I will add a WithMultiEntity property with the logic you are expecting